src/universal/utils/swarm/MediaSwarm.ts

```diff
@@ -0,0 +1,213 @@
+import FastRTCSwarm, {FastRTCSwarmEvents, SwarmConfig} from '@mattkrick/fast-rtc-swarm'
+import {Events, Trebuchet} from '@mattkrick/trebuchet-client'
+import {Dispatch, ReducerAction} from 'react'
+import {StreamUI} from '../../hooks/useSwarm'
+import {StreamName} from './joinSwarm'
+import reducerSwarm from './reducerSwarm'
+
+export interface LocalQualityDict {
+  low: MediaStream | null
+  med: MediaStream | null
+  high: MediaStream | null
+}
+
+const qualityDict: LocalQualityDict = {low: null, med: null, high: null}
+
+interface Config extends SwarmConfig {
+  dispatchState: Dispatch<ReducerAction<typeof reducerSwarm>>
+  trebuchet: Trebuchet
+  userId: string
+}
+
+const qualityConstraints = {
+  audioOnly: {audio: true, video: false},
+  low: {audio: true, video: {width: 64, height: 64}},
+  lowVideo: {audio: false, video: {width: 64, height: 64}}
+}
+
+export default class MediaSwarm extends FastRTCSwarm {
+  trebuchet: Trebuchet
+  dispatchState: Dispatch<ReducerAction<typeof reducerSwarm>>
+  userId: string
+  localStreams = {
+    cam: {...qualityDict},
+    screen: {...qualityDict}
+  }
+
+  constructor (config: Config) {
```

Comment:
Right now, everybody is 64 x 64. In the (near) future, I imagine the person with focus will be some multiple higher than that. How will it work? Will one peer send an event to another saying "please give me your hq" stream?

Reply:
exactly right! a user creates up to 3 streams and serves up the appropriate stream to the peer.